src/lib/components/ui/progress.svelte

```diff
@@ -0,0 +1,75 @@
+<script lang="ts">
+	import type { HTMLAttributes } from 'svelte/elements'
+
+	type Dimension = 'default' | 'large' | 'compact' | 'small'
+	interface Props extends HTMLAttributes<HTMLDivElement> {
+		value?: number
+		dimension?: Dimension
+		class?: string
+	}
+	let {
+		value = 0,
+		dimension = 'default',
+		class: className = '',
+		children,
+		...restProps
+	}: Props = $props()
+</script>
+
+<div class="root {dimension} {className}" {...restProps}>
+	{#if children}
+		{@render children()}
+	{/if}
+	<span class="progress-container">
+		<span class="progress-bar" style={`width: ${value}%`}></span>
+	</span>
+	<div class="value">
+		{value}
+	</div>
+</div>
+
+<style lang="postcss">
+	.root {
+		display: flex;
+		flex: 1;
+		align-items: center;
+		align-self: stretch;
+		gap: 0.5rem;
```

Comment:
Please use variables so that this can be customised.

```suggestion
		gap: var(--half-padding);
```